test/jdk/java/security/KeyStore/CheckMacOSKeyChainIntermediateCATrust.java

```diff
@@ -0,0 +1,192 @@
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.nio.file.Path;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.cert.X509Certificate;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.stream.StreamSupport;
+
+import jdk.test.lib.process.ProcessTools;
+
+import static org.junit.jupiter.api.Assertions.fail;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+/*
+ * @test
+ * @bug 8347067
+ * @library /test/lib
+ * @requires os.family == "mac"
+ * @summary Check whether loading of certificates from macOS Keychain correctly
+ *          loads intermediate CA certificates
+ * @run junit/manual CheckMacOSKeyChainIntermediateCATrust
+ */
+public class CheckMacOSKeyChainIntermediateCATrust {
+
+    private static final String DIR = System.getProperty("test.src", ".");
+
+    @Test
+    public void test() throws Throwable {
+        KeyStore ks = KeyStore.getInstance("KeychainStore");
+        ks.load(null, null);
+
+        Iterator<String> iterator = ks.aliases().asIterator();
+        List<X509Certificate> certificates = StreamSupport.stream(
+                        Spliterators.spliteratorUnknownSize(iterator, Spliterator.ORDERED), false)
+                .sorted()
+                .map(alias -> {
+                    try {
+                        return (X509Certificate) ks.getCertificate(alias);
+                    } catch (KeyStoreException e) {
+                        throw new RuntimeException(e);
+                    }
+                })
+                .toList();
+
+        System.out.println("Verifying expected certificates are trusted");
+
+        String rootCASubjectName = "CN=Example CA,O=Example,C=US";
+        assertThat(containsSubjectName(certificates, rootCASubjectName), "Root CA not found " + rootCASubjectName, certificates);
+
+        String intermediateCASubjectName = "CN=Example Intermediate CA,O=Example,C=US";
+        assertThat(containsSubjectName(certificates, intermediateCASubjectName), "Intermediate CA not found " + intermediateCASubjectName, certificates);
+
+        String nonTrustedCASubjectName = "CN=Non Trusted Example CA,O=Example,C=US";
+        assertThat(not(containsSubjectName(certificates, nonTrustedCASubjectName)), "Non trusted CA found " + nonTrustedCASubjectName, certificates);
```

Comment:
Could you please add one more test, if you don't mind? 

NonTrustedIntermediateCA is issued by nonTrustedCA:
```
openssl genrsa -out non-trusted-intermediate.key 2048
openssl req -new -sha256 -nodes -key non-trusted-intermediate.key  \
  -subj "/C=US/O=Example/CN=Non Trusted Example Intermediate CA" -out non-trusted-intermediate-ca.csr

openssl x509 -req \
 -extensions v3_ca \
 -extfile openssl.cnf \
 -in non-trusted-intermediate-ca.csr \
 -CA non-trusted-test-ca.pem \
 -CAkey non-trusted-root.key \
 -CAcreateserial \
 -out non-trusted-intermediate-ca.pem \
 -days 3650 \
 -sha256
```
In this case, we'll cover all basic scenarios.